backend/imagekit/imagekit.go

```diff
@@ -0,0 +1,794 @@
+	}
+
+	return resp.Body, nil
+}
+
+// Update in to the object with the modTime given of the given size
+//
+// When called from outside an Fs by rclone, src.Size() will always be >= 0.
+// But for unknown-sized objects (indicated by src.Size() == -1), Upload should either
+// return an error or update the object properly (rather than e.g. calling panic).
+func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) error {
+	fs.Debugf(o.remote, "Updating file")
+
+	srcRemote := o.Remote()
+
+	remote := path.Join(o.fs.root, srcRemote)
+	folderPath := o.fs.EncodePath(remote[:strings.LastIndex(remote, "/")+1])
+	fileName := o.fs.EncodeFileName(remote[strings.LastIndex(remote, "/")+1:])
+
+	UseUniqueFileName := new(bool)
+	*UseUniqueFileName = false
+
+	_, res, err := o.fs.ik.Upload(ctx, in, client.UploadParam{
+		FileName:          fileName,
+		Folder:            folderPath,
+		UseUniqueFileName: UseUniqueFileName,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	fileId := res.FileId
+
+	_, file, err := o.fs.ik.File(ctx, fileId)
+
+	if err != nil {
+		return err
+	}
+
+	o.file = *file
+
+	return nil
+}
+
+// Removes this object
+func (o *Object) Remove(ctx context.Context) error {
+
+	fs.Debugf(o.remote, "Removing file: %s", o.file.FileId)
+
+	_, e := o.fs.ik.DeleteFile(ctx, o.file.FileId)
+
+	if e != nil {
+		return e
+	}
+
+	return nil
+}
+
+// SetModTime sets the metadata on the object to set the modification date
+func (o *Object) SetModTime(ctx context.Context, t time.Time) error {
+	return fs.ErrorCantSetModTime
+}
+
+func uploadFile(ctx context.Context, f *Fs, in io.Reader, srcRemote string, options ...fs.OpenOption) (fs.Object, error) {
+
+	fs.Debugf(srcRemote, "Uploading file")
+
+	remote := path.Join(f.root, srcRemote)
+	folderPath := f.EncodePath(remote[:strings.LastIndex(remote, "/")+1])
+	fileName := f.EncodeFileName(remote[strings.LastIndex(remote, "/")+1:])
+
+	UseUniqueFileName := new(bool)
+	*UseUniqueFileName = false
+
+	_, _, err := f.ik.Upload(ctx, in, client.UploadParam{
+		FileName:          fileName,
+		Folder:            folderPath,
+		UseUniqueFileName: UseUniqueFileName,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return f.NewObject(ctx, srcRemote)
+}
+
+func (o *Object) readMetaData(ctx context.Context) error {
+
+	if o.meta != nil {
+		return nil
+	}
+
+	meta := make(map[string]string)
+
+	meta["btime"] = o.file.CreatedAt.Format(time.RFC3339)
+
+	meta["size"] = strconv.FormatUint(o.file.Size, 10)
+
+	meta["fileType"] = string(o.file.FileType)
+	meta["height"] = strconv.Itoa(o.file.Height)
+	meta["width"] = strconv.Itoa(o.file.Width)
+	meta["hasAlpha"] = strconv.FormatBool(o.file.HasAlpha)
+
+	for k, v := range o.file.EmbeddedMetadata {
+		switch v := v.(type) {
+		case string:
+			meta[k] = v
+		case float64:
+			meta[k] = strconv.FormatFloat(v, 'f', -1, 64)
+		case bool:
+			meta[k] = strconv.FormatBool(v)
+		default:
+			meta[k] = fmt.Sprintf("%v", v)
```

Comment:
`fmt.Sprint(v)` is more efficient. It would probably do for all the types and make the type switch redundant.